File: Luiz_Torres/StarWarsShop/app/src/main/java/com/luiztorres/stone/starwarsshop/fragments/CartFragment.java
package com.luiztorres.stone.starwarsshop.fragments;

import android.app.Fragment;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.luiztorres.stone.starwarsshop.R;
import com.luiztorres.stone.starwarsshop.adapters.CartListAdapter;
import com.luiztorres.stone.starwarsshop.extras.CartManager;
import com.luiztorres.stone.starwarsshop.extras.ScreenManager;

/**
 * Created by Dindal on 31/12/2016.
 */

public class CartFragment extends Fragment{
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((AppCompatActivity)getActivity()).getSupportActionBar().setSubtitle(R.string.subtitle_cart);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_cart, container, false);
        return view;
    }

    @Override
    public void onStart() {
        super.onStart();
        CartManager.getInstance(getActivity()).FeedList();

        RelativeLayout im_btn_check = (RelativeLayout) getActivity().findViewById(R.id.btn_cart_checkout);
        im_btn_check.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ScreenManager.getInstance(getActivity()).Replace(new CheckoutFragment());
            }
        });
    }
}

File: app/src/main/java/com/partiufast/mercadodoimperador/ui/fragments/CartFragment.kt
package com.partiufast.mercadodoimperador.ui.fragments

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import com.partiufast.mercadodoimperador.Product
import com.partiufast.mercadodoimperador.R
import com.partiufast.mercadodoimperador.adapters.CartListAdapter
import com.partiufast.mercadodoimperador.ui.activities.PaymentActivity
import java.math.BigDecimal


class CartFragment : Fragment() {
    companion object {
        private val ARG_CART_LIST = "ARG_CART_LIST"
        private var cartProducts: java.util.ArrayList<Product>? = null
        private var recyclerView: RecyclerView? = null
        private var recyclerViewLinearLayout: LinearLayout? = null
        private var emptyCartLinearLayout: LinearLayout? = null

        fun newInstance(cartProducts: ArrayList<Product>): CartFragment {
            val fragment = CartFragment()
            val args = Bundle()
            args.putParcelableArrayList(ARG_CART_LIST, cartProducts)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            cartProducts = arguments.getParcelableArrayList<Product>(ARG_CART_LIST)
        }
    }


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val rootView = inflater!!.inflate(R.layout.cart_fragment, container, false)
        recyclerView = rootView.findViewById(R.id.cart_list_recycler_view) as RecyclerView
        recyclerViewLinearLayout = rootView.findViewById(R.id.recycler_view_layout) as LinearLayout
        emptyCartLinearLayout = rootView.findViewById(R.id.button_layout) as LinearLayout
        val emptyCartButton = rootView.findViewById(R.id.empty_cart_button) as Button
        val checkoutButton = rootView.findViewById(R.id.checkoutButton) as Button

        recyclerView!!.layoutManager = LinearLayoutManager(context)
        recyclerView!!.adapter = CartListAdapter(cartProducts!!)

        val tabLayout = activity.findViewById(R.id.tabs) as TabLayout
        emptyCartButton.setOnClickListener {
            tabLayout.getTabAt(0)?.select()
        }

        checkoutButton.setOnClickListener{
            var total_value = BigDecimal(0)
            for(index in 0..cartProducts!!.size-1) {
                Log.d("item_value", cartProducts!!.get(index).price.toPlainString())
                total_value = total_value.add(cartProducts!!.get(index).price.multiply(BigDecimal(cartProducts!!.get(index).productCount)))
            }

            val intent = Intent(context, PaymentActivity::class.java)
            intent.putExtra(getString(R.string.CART_VALUE_EXTRA), total_value.toString())
                    .putExtra(getString(R.string.CART_VALUE_PLAIN_EXTRA), total_value.toPlainString())
            startActivity(intent)
        }

        return rootView
    }

    fun refreshList() {
        recyclerView!!.adapter.notifyDataSetChanged()
    }

    fun updateListVisibility() {
        if (cartProducts?.size!! > 0) {
            if (recyclerViewLinearLayout?.visibility == View.INVISIBLE) {
                recyclerViewLinearLayout?.visibility = View.VISIBLE
                emptyCartLinearLayout?.visibility = View.INVISIBLE
            }
        } else {
            if (recyclerViewLinearLayout?.visibility == View.VISIBLE) {
                recyclerViewLinearLayout?.visibility = View.INVISIBLE
                emptyCartLinearLayout?.visibility = View.VISIBLE
            }
        }
    }


}
